packages/paste-core/primitives/combobox/stories/index.stories.tsx

```diff
@@ -202,11 +207,28 @@ export const BasicMultiCombobox: React.FC = () => {
     getToggleButtonProps,
     highlightedIndex,
     isOpen,
-    selectedItem,
   } = useComboboxPrimitive({
+    inputValue,
     items: filteredItems,
-    onSelectedItemChange: (args) => {
-      handleSelectItemOnClick(args);
+    onStateChange: (args) => {
+      switch (args.type) {
+        case useComboboxPrimitive.stateChangeTypes.InputChange:
+          if (args.inputValue) {
+            setInputValue(args.inputValue);
+          }
+          break;
+        case useComboboxPrimitive.stateChangeTypes.InputKeyDownEnter:
+        case useComboboxPrimitive.stateChangeTypes.ItemClick:
+        case useComboboxPrimitive.stateChangeTypes.InputBlur:
+          if (args.selectedItem) {
+            handleSelectItemOnChange(args.selectedItem);
+          } else if (args.inputValue) {
```

Comment:
Could you explain the conditional logic here? I'm mostly curious why `handleSelectItemOnChange` is being passed either the selected item object or the input value.

Reply:
@shleewhite Is this the final version of this file? I think I mentioned before, AFAIK `selectedItem` is an object, and I don't think they are interchangable.